server/db/helper.js

```diff
@@ -67,9 +68,31 @@ const getCourse = (id) => {
     });
 };
 
+const getUserBadges = (userId) => {
+  return UserBadge.findAll({
+    where: {
+      id: userId,
+    },
+    
+    attributes: ['idBadge'],
+  })
+  .then((badge) => {
+    return badge.forEach((Badge.findAll({
```

Comment:
You don't need to do a for each, instead you can use the array of badge ids as the specific condition for the findAll
like so - where: { id: badgeIdArray }